Lib9c.Benchmarks/Program.cs

```diff
@@ -0,0 +1,201 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using Bencodex.Types;
+using Libplanet;
+using Libplanet.Action;
+using Libplanet.Assets;
+using Libplanet.Blockchain;
+using Libplanet.Blockchain.Policies;
+using Libplanet.Blocks;
+using Libplanet.RocksDBStore;
+using Libplanet.Store;
+using Libplanet.Store.Trie;
+using Nekoyume.BlockChain;
+using Serilog;
+using Serilog.Events;
+using NCAction = Libplanet.Action.PolymorphicAction<Nekoyume.Action.ActionBase>;
+
+namespace Lib9c.Benchmarks
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            if (args.Length < 2)
+            {
+                Console.Error.WriteLine("Too few arguments.");
+                Environment.Exit(1);
+                return;
+            }
+
+            string storePath = args[0];
+            int limit = int.Parse(args[1]);
+            ILogger logger = new LoggerConfiguration().CreateLogger();
+            Libplanet.Crypto.CryptoConfig.CryptoBackend = new Secp256K1CryptoBackend<SHA256>();
+            var policySource = new BlockPolicySource(logger, LogEventLevel.Verbose);
+            IBlockPolicy<NCAction> policy =
+                policySource.GetPolicy(BlockPolicySource.DifficultyBoundDivisor + 1, 0);
+            var store = new RocksDBStore(storePath);
+            if (!(store.GetCanonicalChainId() is Guid chainId))
+            {
+                Console.Error.WriteLine("There is no canonical chain: {0}", storePath);
+                Environment.Exit(1);
+                return;
+            }
+
+            if (!(store.IndexBlockHash(chainId, 0) is HashDigest<SHA256> gHash))
+            {
+                Console.Error.WriteLine("There is no genesis block: {0}", storePath);
+                Environment.Exit(1);
+                return;
+            }
+
+            DateTimeOffset started = DateTimeOffset.UtcNow;
+            Block<NCAction> genesis = store.GetBlock<NCAction>(gHash);
+            IKeyValueStore stateRootKeyValueStore = new RocksDBKeyValueStore(Path.Combine(storePath, "state_hashes")),
+                stateKeyValueStore = new RocksDBKeyValueStore(Path.Combine(storePath, "states"));
+            IStateStore stateStore = new TrieStateStore(stateKeyValueStore, stateRootKeyValueStore);
+            var chain = new BlockChain<NCAction>(policy, store, stateStore, genesis);
+            long height = chain.Tip.Index;
+            var blockHashes = limit < 0
+                ? chain.BlockHashes.SkipWhile((_, i) => i < height + limit).ToArray()
+                : chain.BlockHashes.Take(limit).ToArray();
+            Console.Error.WriteLine(
+                "Executing {0} blocks: {1}-{2} (inclusive).",
+                blockHashes.Length,
+                blockHashes[0],
+                blockHashes.Last()
+            );
+            DateTimeOffset blocksLoaded = DateTimeOffset.UtcNow;
```

Comment:
IMO Actually, it doesn't mean all the blocks loaded successfully. It means just block hashes loaded and it can cause confusion when analyzing the time from log. How about loading all the blocks at the next line, or updating the log.

```csharp
            Console.WriteLine("Loading blocks\t{0}", blocksLoaded - started);
```

Reply:
That's a valid point.  I am going to make it to load all blocks onto the memory first so that the actual loading time is estimated.

Reply:
Fixed in afba83cf46858e9bef56fede7eb76e4ac45519fa.